Delmar update

jhewitt12 opened the pull request.

jhewitt12 committed:
made map a third of page and changed the google map

jhewitt12 committed:
removed virtual tour button from body text, changed it to 'See Inside', and placed it below the diagram

jhewitt12 committed:
moved keep me in the loop below map, contact info to the right of it

jhewitt12 committed:
changed custom css to columns

jhewitt12 committed:
Merge branch 'master' into updatemap

jhewitt12 committed:
added the news section to see how it will fit next to maps

jhewitt12 committed:
redesigned the features section, removed the current slideshow section

jhewitt12 committed:
added smaller clock image

jhewitt12 committed:
making news section dynamic

jhewitt12 committed:
adding news admin page, working on dynamic news section

jhewitt12 committed:
made a new page for the application, made a new page for news admin

jhewitt12 committed:
separated application, news admin, and about to their own pages, adjusted menu

jhewitt12 committed:
added header

jhewitt12 committed:
removed jquery scroll from external pages

jhewitt12 committed:
Merge branch 'updatemap' into about

jhewitt12 committed:
added videos, paragraphs, and images to about page

jhewitt12 committed:
working on about page

jhewitt12 committed:
made contact icons bigger

jhewitt12 committed:
working on news section php and database integration

jhewitt12 committed:
added new text to about page, added new images and video

jhewitt12 committed:
resolved conflicts

jhewitt12 committed:
resolved conflicts

jhewitt12 committed:
Merge branch 'master' into updatemap

jhewitt12 committed:
finished up first version of news management

jhewitt12 committed:
separated news js into their own files, news section is connected to database, bug fixes on news management

jhewitt12 committed:
added news sql

jhewitt12 committed:
deleted large photos

jhewitt12 committed:
moved video to vimeo and deleted it from assets

jhewitt12 committed:
fixed quick bug

jhewitt12 committed:
assets/vid/IIB 10-06-16 Segment 5A.mp4 to .gitignore

jhewitt12 committed:
rearranged some files

jhewitt12 committed:
fixed scss

jhewitt12 commented:
The scss compiles correctly now and the hero unit is back to normal. Could you put this version on the dev server more me to test? Thanks!

elliottwilliams commented:
Yep. Just so you know, adding commits to a branch adds them to the open pull request, so you'll confuse people if you close a PR and immediately open a new one :)

elliottwilliams commented:
https://github.com/matchboxcowork/matchboxstudio.org/pull/31/commits/81b10de6db105c0f00e1f4ae6ef888d519cb66c0 is live at http://104.131.179.41/ as of `Fri Jan 13 16:05:34 EST 2017`.

elliottwilliams commented:
> Things are looking good overall except it can't find newsCtl.php for some reason. Any ideas? It says this file is not found "http://104.131.179.41/newsCtl.php". But its in the top level of the build folder. Do I need to reference it to somewhere else? Getting close!

It should be getting picked up by guard when building, since it's a `.haml` file in the root directory. I'm taking a look at this, will keep you updated.

elliottwilliams commented:
The issue had to do with Guard (the runner that compiles all dependencies) trying to compile stuff from the `vendor` directory, which is created on production when I ran `bundle install --deployment`. Guard would fail to compile a haml file from within `vendor`, which silently aborted the haml stage of the build.

I pushed a fix to `master` to take care of it. To add it to your branch, do a rebase (`git rebase master`) and a force push (`git push origin delmar_update --force`). Force push is necessary since the rebase modifies commit history.

jhewitt12 committed:
Merge branch 'master' of https://github.com/matchboxcowork/matchboxstudio.org

jhewitt12 committed:
resolved conflicts

jhewitt12 committed:
trying to figure out sql error

jhewitt12 commented:
For some reason its saying the news table doesn't exist in the database when on the home page and news-admin.html page. I never had this issue before. When the page loads, there's an ajax call to newsCtl.php that retrieves the news articles. Did something you changed effect it? I checked my local dev_db.sqlite file and the table is in there. Any ideas? Sorry to be so difficult. I had it all working locally and didn't expect these problems when getting it on the server.

elliottwilliams commented:
No, I was just noticing the same thing. It looks like you changed the dev_db path in https://github.com/matchboxcowork/matchboxstudio.org/pull/31/commits/12f3d3f6ea0b0521c85269ec6336665cd56ab0f9, was that intentional? Changing it back fixed things for me.

jhewitt12 committed:
changed db path, commented out some logs, re-removed mbx.css

jhewitt12 commented:
I remember changing that a long time ago to get it to work. I changed it back and it worked for me too. Okay fingers crossed thats its good now. Could you update the dev server with the latest commit? Thanks!

elliottwilliams commented:
Sure. https://github.com/matchboxcowork/matchboxstudio.org/pull/31/commits/fdd89c174b9ab6dbdd87b363bd95e1e53508fe45 is live on http://104.131.179.41/ as of `Thu Jan 19 16:05:14 EST 2017`.

elliottwilliams commented:
Since it seems like you've got a build working, I'm going to leave some suggestions in your code while we wait to hear back from Jason. Once you have an okay from him, let me know and I'll go about merging and deploying.

elliottwilliams commented on .gitignore:
```diff
@@ -5,3 +5,4 @@ config/prod.php
 dev_db.sqlite3
 build/
 .sass-cache
+assets/vid/IIB 10-06-16 Segment 5A.mp4
```
What's going on here? If this video doesn't need to exist in your source tree, it shouldn't be in the repo. Did you at one point try to upload the video before deciding to use vimeo?

jhewitt12 replied:
Originally it was in the repo but then I realized it was too large to upload to Github. So I deleted it from the repo and had Mikel upload it to Vimeo. I tried to push to Github again and it still gave me errors. Apparently it was still in the history of commits and I had all kinds of trouble trying to remove it from the history. After surfing through forums I finally found the right command and removed it. I'll take that out of the .gitignore.

jhewitt12 commented:
Feel free to leave some suggestions. I'm getting an error with newsCtl.php on http://104.131.179.41/. When I got to  http://104.131.179.41/newsCtl.php I see this error `Fatal error: Uncaught exception 'PDOException' with message 'could not find driver' in /var/www/matchboxstudio.org/build/newsCtl.php:8 Stack trace: #0 /var/www/matchboxstudio.org/build/newsCtl.php(8): PDO->__construct('sqlite:/var/www...', '', '') #1 {main} thrown in /var/www/matchboxstudio.org/build/newsCtl.php on line 8`. Do you get the same thing? How come it works locally but not live?

elliottwilliams commented on create_news.sql:
```diff
@@ -0,0 +1,9 @@
+    description TEXT (1000),
+    active      BOOLEAN,
+    deleted     BOOLEAN
+);
```
Let's keep things organized to accomodate for future changes to the database. Could you:

- rename this file to include a `YYYY-MM-dd` date? i.e. `2017-01-19_create_news.sql`
- move this file to a new `migrations` directory

That way, as new DB changes come in it'll be easier for us to keep track of what has and hasn't been applied.

elliottwilliams commented on news-admin.haml:
```diff
@@ -0,0 +1,93 @@
+
+    / JS
+    =include 'partials/body_scripts.haml'
+
```
Could you move this page and `newsCtl.php.haml` into the `admin` directory along with the other management-related pages?

elliottwilliams commented on newsCtl.php.haml:
```diff
@@ -0,0 +1,108 @@
+    getArticles($dbh);
+
+  }
+  
```
See above comment.

elliottwilliams commented on partials/pricing_and_application.haml:
```diff
@@ -91,7 +91,7 @@
 .container
   %h2.sixteen.columns.message-block
-    %a.button.mbxred{:href => "#app"}
+    %a.button.mbxred{:href => "/about.html"}
```
Did you mean to link to `/apply.html`?

elliottwilliams commented on sass/_imported.scss:
```diff
@@ -251,5 +432,11 @@ img.fullimage {
+    .about-paragraphs .eight:nth-child(2) {
+        padding-left:0;
+    }
+}
```
Sorry I hadn't been clearer about this, but could you move all about-page related scss into its own file?`_imported.scss` contains holdover from the original template, and as you can tell it's pretty unorganized.

Instead, create something like `_about.scss` in the `sass` directory, then add another import statement at the bottom of `mbx.scss`. The underscore in the filename tells Sass not to compile a standalone `about.css`, but rather use it as an import for other files. So importing a stylesheet into `mbx.scss` causes it to end up in the generated `mbx.css`.

elliottwilliams commented:
Looking into it — probably something not installed / configured wrong on the test server.

elliottwilliams commented:
Yeah, it was a missing php module. Should be working now.

elliottwilliams reviewed (changes requested):
Some organizational nitpicks, but overall this looks good. Let me know if I can better clarify any of my comments.

jhewitt12 committed:
completed Elliotts organizational requests

jhewitt12 commented:
I put the news admin pages into the admin folder. I have the username and password to access it but I still get "Server error! Error 500" when trying to load the page so I'm unable to test if it works in its new directory. I'm worried file path relativity may have broken.

elliottwilliams committed:
Using absolute paths on partials and asset hrefs to support pages in a subdirectory.

This fixes problems with `admin/news-admin.haml` and `admin/newsCtl.php.haml`.

elliottwilliams commented:
See the commit I added to use absolute paths in partials and on those header pages.

https://github.com/matchboxcowork/matchboxstudio.org/pull/31/commits/ded2cbc961eb24a2723c7e723cbd2d88512f08df is now live on http://104.131.179.41/ as of `Fri Jan 20 10:10:37 EST 2017`.

jhewitt12 commented:
That looks good on the server! I noticed an image file name needed to be changed from .JPG to .jpg on the about.html page so I pulled your commit. It broke everything locally. Do I need to run a rake command or something? Basically it can't find any of the files in assets folder.

elliottwilliams commented:
`rake clean` followed by `rake` will delete the `build` directory and re-create it, try that.

jhewitt12 commented:
I did that and its still broken. Any other ideas?

elliottwilliams commented:
How are you serving your dev site? Make sure you're using the server that `guard` starts up (`localhost:9292`). If your server doesn't have `/` as the site's root, then this commit will break asset paths for you. Additionally, verify that you can get an asset from its URL (like `/assets/img/Rendering.png`)

jhewitt12 committed:
fixed img extension

jhewitt12 commented:
When I run it using XAMPP it doesn't work because I had localhost/matchboxstudio.org/. It does work when I serve it without XAMPP. I pushed the img ext change. Other than that I think it all looks good.

elliottwilliams commented:
Okay, awesome. 
https://github.com/matchboxcowork/matchboxstudio.org/pull/31/commits/026c78925208a4c8a7abff0f856ae84ab1b62e0b is now live on http://104.131.179.41/ as of `Fri Jan 20 11:36:18 EST 2017`.

I'm going to assign @mikelberger as a reviewer to the pull request. Once he looks at the site and approves his review, let's merge and deploy.

jhewitt12 commented:
Sounds great

mikelberger reviewed (commented):
https://www.dropbox.com/s/9dxlq93xb351bcu/Screenshot%202017-01-20%2014.01.55.png?dl=0

Getting an HTTPS error.

elliottwilliams committed:
Revert removing https prefix from https-only resources.

elliottwilliams commented:
Good catch, I reverted my change that removed `https` prefixed in a couple asset URLs.

elliottwilliams commented:
https://github.com/matchboxcowork/matchboxstudio.org/pull/31/commits/c5aad4b29dedc6782afab690751553dc0cb5ccd7 live on http://104.131.179.41 as of `Fri Jan 20 14:27:27 EST 2017`.

jhewitt12 committed:
changed google map to https

jhewitt12 committed:
Merge branch 'delmar_update' of https://github.com/matchboxcowork/matchboxstudio.org into delmar_update

jhewitt12 committed:
made 'See inside' button toggle to close tour, moved 'See Inside' button below tour, changed red divider text to blue, adding padding around contact section

jhewitt12 committed:
fixed 'see inside' button bug

jhewitt12 committed:
made pricing its own partial and included it on the join page

elliottwilliams committed:
Normalized contrast across sponsor logos.

A script `normalize.sh` is included, to re-perform the normalization on all the
logos. ImageMagick is required.

See <http://www.imagemagick.org/Usage/color_mods/#normalize> for a description
of the normalization operation.

elliottwilliams committed:
Adjusted the opacities used to show sponsor logos.

elliottwilliams commented:
Here's a task list based on Jason's review of the design changes. I've tried to check off what's already been done, please check off others as you complete them:

- [x] Show pricing information on the application screen
- [x] Remove separate "pricing" and "apply" tabs from main nav
- [x] Add a full-width bar of white space between the street view unit and the background behind it
- [x] Toggle buttons to open/close the street view unit
- [x] Add padding between the map/news row and the contact information, and add padding between contact information and sponsorship rows
- [x] Normalize the colors in sponsor logos and lighten the overall greys
- [x] Changing word-accent color to navy blue
- [x] Title case the byline on the about page
- [x] Change the "+" tab to show testimonies in a single column
  - I need written testominies to proceed here -@elliott

elliottwilliams committed:
Removed superfluous hr above the sponsors unit and replaced it with an increased margin

elliottwilliams commented:
http://104.131.179.41 is live at https://github.com/matchboxcowork/matchboxstudio.org/pull/31/commits/adc121394020dbafca2857160cc80ce2507c48f6 as of `Sun Jan 29 15:03:07 EST 2017`.

elliottwilliams committed:
Added member testimonians to slide panel.

elliottwilliams commented:
I updated the testimonials to what @amandafindlay sent me. You can see the preview at 
http://104.131.179.41/ (showing `4e58e79` as of `Mon Feb 20 22:57:00 EST 2017`).

@mikelberger, can you sign off on this so we can deploy? Thanks!

elliottwilliams committed:
Revert change to emphasized text color: back to red.

elliottwilliams committed:
Changed "See Inside" button to "Explore Inside"

elliottwilliams committed:
Changed apply button to "Join" in the pricing unit, and better aligned it.

elliottwilliams committed:
Removed "See pricing info" link from application form.

elliottwilliams committed:
Navigation to anchors and links from the mobile menu.

elliottwilliams committed:
Updated Austin's testimony

elliottwilliams committed:
Added new sponsor logos

elliottwilliams committed:
Resized and converted new logos to grayscale.

elliottwilliams committed:
Updated Stuart and Branigin's logo.

elliottwilliams committed:
Rebuild sponsors partial when sponsors.yaml changes.

elliottwilliams committed:
Added new sponsors to sponsors.yaml

elliottwilliams committed:
Hovering the social badges now evenly darkens them.

elliottwilliams committed:
Updated font-awesome and added instagram icon.

elliottwilliams committed:
Shrunk hero logo size on mobile portrait.

elliottwilliams committed:
Vertically grouped social media icons.
